packages/derived-wallet-ethereum/src/EIP1193DerivedSignature.ts

```diff
@@ -6,39 +6,54 @@ import {
   Signature,
 } from "@aptos-labs/ts-sdk";
 
-export class EIP1193DerivedSignature extends Signature {
+/**
+ * A classs representing a signature of a message signed with EIP1193
+ */
+export class EIP1193Signature extends Signature {
   static readonly LENGTH = 65;
+  protected readonly _siweSignature: Uint8Array;
 
-  // The signature of the message
-  private readonly _siweSignature: Uint8Array;
+  constructor(siweSignature: HexInput) {
+    super();
+    this._siweSignature = Hex.fromHexInput(siweSignature).toUint8Array();
+  }
+
+  get siweSignature() {
+    return Hex.fromHexInput(this._siweSignature).toString();
+  }
+
+  serialize(serializer: Serializer) {
+    serializer.serializeBytes(this._siweSignature);
+  }
+
+  static deserialize(deserializer: Deserializer) {
+    const signature = deserializer.deserializeBytes();
+    return new EIP1193Signature(signature);
+  }
+}
+
+/**
+ * A class representing a signature of a message signed with EIP1193 and following the SIWE standard
+ */
+export class EIP1193DerivedSignature extends EIP1193Signature {
```

Comment:
why is a second class being defined?
any reason for not using a single class?

Reply:
An `EIP1193Signature` is needed for when [signing a message ](https://github.com/aptos-labs/aptos-wallet-adapter/pull/572/files#diff-a45fd8bd4a291063ae3168c105c1d40f1534684bdc323da28afb774895ab9ab3R72) - this does not need the `issuedAt` and `scheme`.

An `EIP1193DerivedSignature` is needed for signing a transaction message - this requires the `issuedAt` and `scheme`.

We could make both `issuedAt` and `scheme` optional and use one class, but I prefer to make them require as they are required for signing a transaction message.
So We now have 2 classes, one defines an `EIP1193Signature` and another one `EIP1193DerivedSignature` that extends `EIP1193Signature`.

Reply:
Oh, I see.
This is a bit confusing and I wish we had a single class for signatures :(
but I guess it's easier this way at the moment

Reply:
I agree with you that the fields should be required, cause they are for transactions

Reply:
How about renaming them `EIP1193PersonalSignature` and `EIP1193SiweSignature`?
So the distinction is a bit clearer?